lib/interfaces/include/InverterInterface.h

```diff
@@ -1,17 +1,96 @@
 #ifndef INVERTERINTERFACE_H
 #define INVERTERINTERFACE_H
+#include <stdint.h>
 
-#include "shared_types.h"
+#include "FlexCAN_T4.h"
+#include "MessageQueueDefine.h"
 
+#include <hytech.h>
+#include "DrivetrainSystem.h"
+#include <CANInterface.h>
+#include <shared_types.h>
+#include "VCRCANInterfaceImpl.h"
 
-// struct InverterMotorControl_s
-// {
+struct InverterParams_s
+{   
+    float MINIMUM_HV_VOLTAGE; 
+};
 
-// }
+/**
+ * Struct containing id info for this specific inverter interface
+ */
+struct InverterIds_s
+{
+    uint32_t inv_control_word_id; 
+    uint32_t inv_control_input_id; 
+    uint32_t inv_control_parameter_id; 
+};
 
+/**
+ * Inverter interface
+ */
 class InverterInterface
 {
-    public:
-    InverterInterface() = default;
+
+    public: 
+
+        InverterInterface(
+            CANBufferType *msg_output_queue, 
+            uint32_t inv_control_word_id,
+            uint32_t inv_control_input_id,
+            uint32_t inv_control_params_id,
+            InverterParams_s inverter_params) : msg_queue_(msg_output_queue), _inverter_params(inverter_params)
+        { 
+            inverter_ids.inv_control_word_id = inv_control_word_id;
+            inverter_ids.inv_control_parameter_id = inv_control_params_id;
+            inverter_ids.inv_control_input_id = inv_control_input_id;
+        }
+
+        // TODO un-public these (they are public for testing)
```

Comment:
shouldn't these stay public? Unless there's something else that used to be public that's since been fixed